### DISCUSSION TITLE
choosing between rsl_rl, rl_games, and skrl

### DISCUSSION THREAD

**Discussion by dxyy1** [isaac-sim/IsaacLab#1398] [Q&A] [answered]
Hi there,

I was wondering which training algorithm package I should pick for my research, out of rsl_rl, rl_games, and skrl?

I am quite new to the field of robotics and currently working on legged robots, specifically locomotion. I summarised my situation below. I am:
1. currently developing my own neural network models, such as hierachical models;
2. expecting to do a lot of rewards tuning, tweaking of the training algorithms, etc;
3. likely to only use PPO, unless there is a clear better choice? (any opinion on this is welcomed);
4. seeing that most published code on legged locomotion are done using rsl_rl, so atm I am leaning towards it.
5. going to be deploying the model to various unitree robots, so I will be concerned with sim-to-real transfer. 

I'd like to know everyone's opinion on this: how's it been like to use any one of those packages, did it had any issue during deployment for legged robots (quadrupeds, humanoid, ...), which one you'd recommend if I am starting anew?

Any information will be more than helpful.
Thanks in advance :) 

![image](https://github.com/user-attachments/assets/e6904a86-e8fb-4ff3-837d-7d289d1500c6)

**Comment by Toni-SM** [comment, marked as answer]
Hi @dxyy1 

Please see https://isaac-sim.github.io/IsaacLab/main/source/overview/reinforcement-learning/rl_frameworks.html to have a better reference of the different supported RL libraries
